Source/WebCore/platform/graphics/avfoundation/MediaPlaybackTargetCocoa.h

```diff
@@ -28,34 +28,38 @@
 #if ENABLE(WIRELESS_PLAYBACK_TARGET)
 
 #include "MediaPlaybackTarget.h"
-#include <wtf/RetainPtr.h>
+#include "MediaPlaybackTargetContextCocoa.h"
+
+OBJC_CLASS AVOutputContext;
 
 namespace WebCore {
 
 class MediaPlaybackTargetCocoa : public MediaPlaybackTarget {
 public:
-    WEBCORE_EXPORT static Ref<MediaPlaybackTarget> create(AVOutputContext *);
-    WEBCORE_EXPORT static Ref<MediaPlaybackTarget> create(MediaPlaybackTargetContext&&);
+    WEBCORE_EXPORT static Ref<MediaPlaybackTarget> create(MediaPlaybackTargetContextCocoa&&);
 
 #if PLATFORM(IOS_FAMILY) && !PLATFORM(IOS_FAMILY_SIMULATOR) && !PLATFORM(MACCATALYST)
-    static Ref<MediaPlaybackTarget> create();
+    WEBCORE_EXPORT static Ref<MediaPlaybackTargetCocoa> create();
 #endif
 
-    virtual ~MediaPlaybackTargetCocoa();
-
     TargetType targetType() const final { return TargetType::AVFoundation; }
     const MediaPlaybackTargetContext& targetContext() const final { return m_context; }
 
+    AVOutputContext* outputContext() { return m_context.outputContext().get(); }
+
 protected:
-    explicit MediaPlaybackTargetCocoa(AVOutputContext *);
```

Comment:
Why protected, can we make all of them private?
Should MediaPlaybackTargetCocoa be final?

Reply:
we can; visibility was all existing code. I don't tend to modif what was already in place like that.